app/models/spree/variant.rb

```diff
@@ -170,6 +170,11 @@ def self.active(currency = nil)
                                           select("spree_variants.id") })
     end
 
+    def self.linked_to(semantic_id)
+      includes(:semantic_links).references(:semantic_links)
+        .where(semantic_links: { semantic_id: }).first
```

Comment:
I think that `semantic_id` is pulled the object's Id in the graph - that will always be a URI, so should be all good! 🤞🏻

Reply:
If a user imports a product several times, or several users import the same product and share a catalog then we can have multiple OFN products linked to the same semantic id. And I don't think that we should restrict that. Otherwise only one OFN shop would be able to sell a linked DFC product. But we want several regional food hubs selling from one wholesaler.

It's still possible for a shop to sell several products linked to the same wholesale variant in one order cycle but that's very unlikely. The only thing that happens then is that we may not reduce the wholesale order enough after retail orders have been cancelled. But we can solve that later if that's actually a problem.